sdk/test/UnitTests/Custom/Runtime/UserAgentTests.cs

```diff
@@ -308,5 +357,29 @@ public override void InvokeSync(IExecutionContext executionContext)
                 LastRequest = executionContext.RequestContext.Request;
             }
         }
+
+        public class TempMeterProvider : MeterProvider
+        {
+            public override Meter GetMeter(string scope, Attributes attributes = null)
+            {
+                var assembly = typeof(MeterProvider).Assembly;
+                var internalType = assembly.GetType("Amazon.Runtime.Telemetry.Metrics.NoOp.NoOpMeter", throwOnError: true);
```

Comment:
No, let's not do this. The `Core` internal types are visible in the unit tests, you should be able to instantiate them directly.

You may need to update the AssemblyInfo to include the `AWSSDK.UnitTests.Custom.NetFramework` project: https://github.com/aws/aws-sdk-net/blob/v4-development/generator/ServiceClientGeneratorLib/Generators/SourceFiles/CoreAssemblyInfo.tt#L27